components/event-feed-service/pkg/integration_test/feed_purge_test.go

```diff
@@ -189,23 +190,41 @@ func TestPurgeServer(t *testing.T) {
 				require.NoError(t, err)
 
 				// wait until the purge job is done or we time out waiting for it
-				ticker := time.Tick(1 * time.Second)
+				checkPurgeStartEnd := func() error {
+					status, err := testSuite.purgeClient.Show(context.Background(), &data_lifecycle.ShowRequest{})
+					if err != nil {
+						return errors.Wrap(err, "failed to get purge show")
+					}
+
+					lastEnd, err := ptypes.Timestamp(status.LastEnd)
+					if err != nil {
+						return errors.Wrapf(err, "failed to parse last end: '%v'", status.LastEnd)
+					}
+
+					if lastEnd.After(startTime) {
+						return nil
+					}
+
+					return errors.Errorf("last end '%v' not after start time '%v'", lastEnd, startTime)
+				}
+
 				func() {
```

Comment:
dont think you need func here

Reply:
It's so I don't have to use GOTO's.

Reply:
Welp, you're right, after I refactored the bulk of the stuff into another func I don't need that one